packages/variable-editor/src/components/variables/data/variable-utils.ts

```diff
@@ -125,3 +133,74 @@ const enrichVariableWithFileMetadata = (variable: Variable, metadata: string) =>
   }
   return variable;
 };
+
+export const toContent = (variables: Array<Variable>) => {
+  return stringify(new Pair(new Scalar('Variables'), parseVariables(variables)));
+};
+
+const parseVariables = (variables: Array<Variable>) => {
+  const map = new YAMLMap();
+  variables.forEach(variable => map.add(parseVariable(variable)));
+  return map;
+};
+
+const parseVariable = (variable: Variable) => {
+  const variableKey = new Scalar(variable.name);
+  variableKey.commentBefore = parseKeyDescription(variable);
+  let variableValue;
+  if (variable.children.length === 0) {
+    variableValue = new Scalar(parseVariableValue(variable.value));
+  } else {
+    variableValue = parseVariables(variable.children);
+  }
+  return new Pair(variableKey, variableValue);
+};
+
+const parseKeyDescription = (variable: Variable) => {
+  const metadata = variable.metadata;
+  const description = variable.description;
+  if (metadata.type === '') {
+    return description && addSingleLeadingWhitespaceToEachLine(description);
+  }
+
+  const metadataComment = parseMetadataComment(metadata);
+  const commentBefore = description ? description + '\n' + metadataComment : metadataComment;
+  return addSingleLeadingWhitespaceToEachLine(commentBefore);
+};
+
+const parseMetadataComment = (metadata: Metadata) => {
+  let metadataComment;
+  switch (metadata.type) {
+    case 'enum':
+      if (isEnumMetadata(metadata)) {
+        metadataComment = 'enum: ' + metadata.values.join(', ');
+      }
+      break;
+    case 'file':
+      if (isFileMetadata(metadata)) {
```

Comment:
Same here

Reply:
Ahh, sorry I had a comment on line 178 which was swallowed. Why are you double checking if the type is file or enum? Isn't it clear with the switch already?

Reply:
This is a type guard function:
```
isFileMetadata = (metadata?: Metadata): metadata is FileMetadata => {...}
```
You can see that it has `metadata is FileMetadata` in its signature. If this function returns true, TypeScript accepts that the argument is of type `FileMetadata` within the scope of the if block. Otherwise, TypeScript would show an error on the next line when trying to access the attribute `values`, since it does not exist on the type `Metadata` but only on its sub-type `FileMetadata`. Essentially, its a save way to cast an object of type `Metadata` to its sub-type `FileMetadata`.

Reply:
Understand. good